backend/database/seeding/test_helpers.go

```diff
@@ -228,22 +230,99 @@ func GetOperationFromSlug(t *testing.T, db *database.Connection, slug string) mo
 	return fullOp
 }
 
-func GetOperations(t *testing.T, db *database.Connection) []models.Operation {
-	var fullOps []models.Operation
-	err := db.Select(&fullOps, sq.Select("id", "slug", "name", "status").
-		From("operations"))
+func GetOperations(t *testing.T, db *database.Connection) []services.OperationWithID {
+	var operations []struct {
+		models.Operation
+		NumUsers    int `db:"num_users"`
+		NumEvidence int `db:"num_evidence"`
+		NumTags     int `db:"num_tags"`
+	}
+
+	var topContribs []dtos.TopContrib
+
+	var evidenceCount []dtos.EvidenceCount
+
+	err := db.Select(&operations, sq.Select("operations.id", "slug", "operations.name", "status", "count(distinct(user_operation_permissions.user_id)) AS num_users", "count(distinct(evidence.id)) AS num_evidence", "count(distinct(tags.id)) AS num_tags").
+		From("operations").
+		LeftJoin("user_operation_permissions ON user_operation_permissions.operation_id = operations.id").
+		LeftJoin("evidence ON evidence.operation_id = operations.id").
+		LeftJoin("tags ON tags.operation_id = operations.id").
+		GroupBy("operations.id").
+		OrderBy("operations.created_at DESC"))
+
 	require.NoError(t, err)
-	return fullOps
+
+	err = db.SelectRaw(&topContribs, services.GetTopContributorsForEachOperation)
+
+	require.NoError(t, err)
+
+	err = db.SelectRaw(&evidenceCount, services.EvidenceCountForAllOperations)
+
+	require.NoError(t, err)
+
+	operationsWithID := []services.OperationWithID{}
+	for _, operation := range operations {
+
+		var topContribsForOp []dtos.TopContrib
+		for i := range topContribs {
+			if topContribs[i].OperationID == operation.ID {
+				topContribsForOp = append(topContribsForOp, topContribs[i])
+			}
```

Comment:
You might be able to use Filter for this. Additionally/Alternatively, if you know the exact size of topContribsForOp, you could pre-initialize the array to the correct length to avoid extra memory allocations, or if you know a top bound, you could pre-initialize the array to have the maximum capacity you need.

This might work using filter:
```go
topContribsForOp := helpers.Filter(topContribs, func(item dtos.TopContrib) {return item.OperationID == operation.ID} )
```

Reply:
I'll give it a shot!

Reply:
yeah same issue here:
> type func(item dtos.TopContrib) of (func(item dtos.TopContrib) literal) does not match inferred type func(dtos.TopContrib) bool for func(T) boolcompiler[CannotInferTypeArgs](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal?utm_source%3Dgopls#CannotInferTypeArgs)

Reply:
Try this:
```go
		_, evidenceCountForOpPtr := helpers.Find(evidenceCount, func(item dtos.EvidenceCount) bool {
			return item.OperationID == operation.ID
		})
```

I think I just forgot to type the return. This usage is slightly harder to use though, because you need to make sure that the value is there before you dereference it, and return a default value otherwise.

Reply:
Oh, that was for find. here's for filter:

```go
               topContribsForOp := helpers.Filter(topContribs, func(contributor dtos.TopContrib) bool {
			return contributor.OperationID == operation.ID
		})
```

Reply:
I also tried to implement the Find method:
```go
		var evidenceCountForOp dtos.EvidenceCount
		_, evidenceCountForOpPtr := helpers.Find(evidenceCount, func(item EvidenceCountWithID) bool {
			return item.OperationID == operation.ID
		})
		evidenceCountForOp.CodeblockCount = evidenceCountForOpPtr.CodeblockCount
		evidenceCountForOp.ImageCount = evidenceCountForOpPtr.ImageCount
		evidenceCountForOp.HarCount = evidenceCountForOpPtr.HarCount
		evidenceCountForOp.EventCount = evidenceCountForOpPtr.EventCount
		evidenceCountForOp.RecordingCount = evidenceCountForOpPtr.RecordingCount
```
instead of the typical loop:
```go
var evidenceCountForOp dtos.EvidenceCount
for i := range evidenceCount {
			if evidenceCount[i].OperationID == operation.ID {
				evidenceCountForOp.CodeblockCount = evidenceCount[i].CodeblockCount
				evidenceCountForOp.ImageCount = evidenceCount[i].ImageCount
				evidenceCountForOp.HarCount = evidenceCount[i].HarCount
				evidenceCountForOp.EventCount = evidenceCount[i].EventCount
				evidenceCountForOp.RecordingCount = evidenceCount[i].RecordingCount
				break
			}
		}
```

but I got this error:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
``` 
any thoughts?

Reply:
TLDR; I want to follow the conventions of the repo, but am running into an error on the `Find` method, and am not sure the `Filter` method is more performant or easier to read; if it's all the same to you I might leave them as I had them, but am also totally open to changing them!

Reply:
changed to use find and filter